### ISSUE TITLE
Popover: Docs update for A11y guidance and close with keyboard decision

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux1#862]
Update popover documentation and example for how to make accessible - 
- Don’t change to insert in body - keep in order following trigger element. Don’t use UI bootstrap popover-append-to-body=true (2.4.3)
- Trigger element needs to be in tab order & receive focus (usable with keyboard)
- Element with hide() function needs to be in tab order and receive focus
- If popover will contain a menu or submenu - have aria-haspopup=true attribute on triggering element

**Decision** - popover must have element to close itself OR we add support for esc key to close with keyboard. Otherwise there is no way to close popover with keyboard

Update demo examples to be accessible for the above guidance
